src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -243,6 +243,39 @@ asio::awaitable<void> LFUDAPolicy::redis_sync(const DoutPrefixProvider* dpp, opt
   }
 }
 
+// Changes state to INVALID for dirty objects. An INVALID state indicates that a delete request has been
+// issued on an object and it must be deleted rather than written to the backend. This lazy deletion occurs
+// in the Cleaning method and prevents data races during concurrent requests. This method returns "false"
+// if the state has not been set to INVALID, and "true" if it has.
```

Comment:
You can add a comment saying that the state is NOT set to INVALID, when cleaning is in progress which writes the object to the backend store.